rust/libnewsboat-ffi/src/utils.rs

```diff
@@ -276,6 +278,32 @@ pub extern "C" fn rs_make_title(input: *const c_char) -> *mut c_char {
     })
 }
 
+#[no_mangle]
+/// Gets the current working directory or an empty string on error.
+pub extern "C" fn rs_getcwd() -> *mut c_char {
+    abort_on_panic(|| {
+        let result = utils::getcwd()
+            .and_then(|path| {
+                path.to_str()
+                    .ok_or(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "Invalid unicode in path",
+                    ))
+                    .map(String::from)
```

Comment:
I didn't find a way to get from `PathBuf` to `Cstring` directly without converting to `String` first.

Reply:
I don't see a way to do this, too. It doesn't look like `utils::getcwd` is used in any performance-sensitive context, so we should be fine :)

Reply:
I did ask on the Rust discord and found out the following:
use std::os::unix::ffi::OsStringExt;
```Rust
fn main() {
    let path_buf = std::path::PathBuf::from("/foo");
    
    let c_string = std::ffi::CString::new(path_buf.into_os_string().into_vec());
    
    println!("{:?}", c_string);
}
```
Windows portability isn't an issue for newsboat right?

Reply:
No, Windows portability isn't important, and will probably never be—I doubt there is a user base there (and they have WSL now anyway, so should be fine without any effort on our part). The code looks pretty fine, if a bit arcane :)